src/main/kotlin/io/polygon/kotlin/sdk/websocket/PolygonWebSocketClient.kt

```diff
@@ -195,46 +213,103 @@ constructor(
-            frame.jsonArray.forEach { processFrameJson(it, collector) }
-        }
-
-        if (frame is JsonObject) {
-            val message = when (frame.jsonObject[EVENT_TYPE_MESSAGE_KEY]?.jsonPrimitive?.content) {
-                "status" -> serializer.decodeFromJsonElement(StatusMessage.serializer(), frame)
-                "T" -> serializer.decodeFromJsonElement(StocksMessage.Trade.serializer(), frame)
-                "Q" -> serializer.decodeFromJsonElement(StocksMessage.Quote.serializer(), frame)
-                "A", "AM" -> serializer.decodeFromJsonElement(StocksMessage.Aggregate.serializer(), frame)
-                "C" -> serializer.decodeFromJsonElement(ForexMessage.Quote.serializer(), frame)
-                "CA" -> serializer.decodeFromJsonElement(ForexMessage.Aggregate.serializer(), frame)
-                "XQ" -> serializer.decodeFromJsonElement(CryptoMessage.Quote.serializer(), frame)
-                "XT" -> serializer.decodeFromJsonElement(CryptoMessage.Trade.serializer(), frame)
-                "XA" -> serializer.decodeFromJsonElement(CryptoMessage.Aggregate.serializer(), frame)
-                "XS" -> serializer.decodeFromJsonElement(CryptoMessage.ConsolidatedQuote.serializer(), frame)
-                "XL2" -> serializer.decodeFromJsonElement(CryptoMessage.Level2Tick.serializer(), frame)
-                "V" -> serializer.decodeFromJsonElement(IndicesMessage.Value.serializer(), frame)
-                "LV" -> serializer.decodeFromJsonElement(LaunchpadMessage.LaunchpadValue.serializer(), frame)
-                else -> RawMessage(frame.toString().toByteArray())
-            }
-            collector.add(message)
-        }
-
-        return collector
-    }
+	@Throws(SerializationException::class)
+	private fun processFrameJson(
+	    market: Market,
+	    feed: Feed,
+	    frame: JsonElement,
+	    collector: MutableList<PolygonWebSocketMessage> = mutableListOf()
+	): List<PolygonWebSocketMessage> {
+
+	    if (frame is JsonArray) {
+	        frame.jsonArray.forEach { processFrameJson(market, feed, it, collector) }
+	    }
+
+	    if (frame is JsonObject) {
+	        val message = when (market) {
+	            Market.Stocks -> parseStockMessage(frame)
+	            Market.Options -> parseOptionMessage(frame)
+	            Market.Indices -> parseIndicesMessage(frame)
+	            Market.Forex -> parseForexMessage(frame)
+	            Market.Crypto -> parseCryptoMessage(frame)
+	            Market.LaunchpadStocks -> parseLaunchpadMessage(frame)
+	            Market.LaunchpadOptions -> parseLaunchpadMessage(frame)
+	            Market.LaunchpadForex -> parseLaunchpadMessage(frame)
+	            Market.LaunchpadCrypto -> parseLaunchpadMessage(frame)
+	        }
+	        val finalMessage = message ?: RawMessage(frame.toString().toByteArray())
+	        collector.add(finalMessage)
+	    }
+
+	    return collector
+	}
+
+	private fun parseStockMessage(frame: JsonObject): PolygonWebSocketMessage? {
```

Comment:
I don't think this or any of the other `parse...` functions need to return a nullable result, since all of them have `else` branches they all return a non-null `PolygonWebsocketMessage`

As an added bonus if all of the parse functions return a non-null result, we don't need that `finalMessage` line that null-coalesces anymore